packages/core/src/TxMetadata/cip20.ts

```diff
@@ -0,0 +1,56 @@
+import { StringUtils } from '@cardano-sdk/util';
+import { TxMetadata } from '../Cardano';
+
+export const CIP_20_METADATUM_LABEL = 674n;
+
+export enum MessageValidationFailure {
+  wrongType = 'WrongType',
+  oversize = 'Oversize'
+}
+
+export type MessageValidationResult = {
+  valid: boolean;
+  failure?: MessageValidationFailure;
+};
+
+export type CIP20TxMetadataMessage = string;
+
+export type CIP20TxMetadataArgs = {
+  // An array of message strings, limited to 64 bytes each
+  messages: CIP20TxMetadataMessage[];
+};
+
+export type ValidationFailureMap = Map<CIP20TxMetadataMessage, MessageValidationResult>;
+
+/**
+ * Validate each message for correct type, in the case of JavaScript, and size constraint
+ *
+ * @param entry unknown
+ * @returns Validation result
+ */
+export const validateMessage = (entry: unknown): MessageValidationResult => {
+  if (typeof entry !== 'string') return { failure: MessageValidationFailure.wrongType, valid: false };
+  if (StringUtils.byteSize(entry) > 64) return { failure: MessageValidationFailure.oversize, valid: false };
+  return { valid: true };
+};
+
+export const isTxMetadata = (
+  value: TxMetadata | Map<CIP20TxMetadataMessage, MessageValidationResult>
```

Comment:
**(not a change req)** a common Result type would be nice to have in the SDK for consistency. There are more places where this would be useful
```suggestion
  value: Result<TxMetadata, Map<CIP20TxMetadataMessage, MessageValidationResult>>
```

Reply:
Actually maybe this is a change req :sweat_smile: Right now it will be difficult to differentiate between error and valid result, as both can be a `Map`